src/main/java/org/interledger/ilqp/QuoteLiquidityResponse.java

```diff
@@ -48,5 +48,148 @@
    * @return An instance of {@link ZonedDateTime}.
    */
   ZonedDateTime getExpiresAt();
+  
+  /**
+   * A builder for instances of {@link QuoteLiquidityResponse}.
+   */
+  class Builder {
+    private LiquidityCurve liquidityCurve;
+    private InterledgerAddress appliesTo;
+    private Duration sourceHoldDuration;
+    private ZonedDateTime expiresAt;
+    
+    /**
+     * Set the liquidity curve into this builder.
+     *
+     * @param liquidityCurve To An instance of {@link LiquidityCurve}.
+     */
+    public Builder liquidityCurve(final LiquidityCurve liquidityCurve) {
+      this.liquidityCurve = Objects.requireNonNull(liquidityCurve);
```

Comment:
Yip agreed - it would be an error to try set a null liquidity curve into the quote response builder - the curve should always at least be empty rather than null. I'll leave this as-is.